devito/dse/aliases.py

```diff
@@ -36,244 +38,246 @@ def collect(exprs):
-        if len(handle) != 1:
-            raise DSEException("%s cannot be represented by the COM %s" %
-                               (str(ofs), str(COM)))
-        distances.append(handle.pop())
+    return Candidate(expr.rhs, indexeds, bases, offsets)
 
-    return COM, distances
 
-
-def calculate_offsets(indexeds):
+def compare_ops(e1, e2):
     """
-    Return a list of tuples, with one tuple for each indexed object appearing
-    in ``indexeds``. A tuple represents the offsets from the origin (0, 0, ..., 0)
-    along each dimension. All objects must use the same indices, in the same
-    order; otherwise, ``None`` is returned.
-
-    For example, given: ::
-
-        indexeds = [A[i,j,k], B[i,j+2,k+3]]
-
-    Return: ::
-
-        [(0, 0, 0), (0, 2, 3)]
+    Return True if the two expressions ``e1`` and ``e2`` perform the same arithmetic
+    operations over the same input operands, False otherwise.
     """
-    reference = indexeds[0]
-    aindices = IterationInstance(reference).aindices
-    zero = IterationInstance(reference.function[aindices])
-    try:
-        offsets = [IterationInstance(i) - zero for i in indexeds]
-    except TypeError:
-        # Vectors have different rank
-        return None
-    if all(i.is_constant for i in offsets):
-        return aindices, offsets
+    if type(e1) == type(e2) and len(e1.args) == len(e2.args):
+        if e1.is_Atom:
+            return True if e1 == e2 else False
+        elif isinstance(e1, Indexed) and isinstance(e2, Indexed):
+            return True if e1.base == e2.base else False
+        else:
+            for a1, a2 in zip(e1.args, e2.args):
+                if not compare_ops(a1, a2):
+                    return False
+            return True
     else:
-        return None
+        return False
 
 
-def is_translated(ofs1, ofs2):
+def is_translated(c1, c2):
     """
-    Return True if ``ofs2`` is translated w.r.t. to ``ofs1``, False otherwise.
+    Given two potential aliases ``c1`` and ``c2``, return True if ``c1``
+    is translated w.r.t. ``c2``, False otherwise.
 
     For example: ::
 
-        e1 = A[i,j] + A[i,j+1]
-        e2 = A[i+1,j] + A[i+1,j+1]
+        c1 = A[i,j] + A[i,j+1]
+        c2 = A[i+1,j] + A[i+1,j+1]
 
-    ``ofs1`` would be [(0, 0), (0, 1)], while ``ofs2`` would be [(1, 0), (1,1)], so
-    ``e2`` is translated w.r.t. ``e1`` by ``(1, 0)``, and True is returned.
+    ``c1``'s Toffsets are ``{i: [0, 0], j: [0, 1]}``, while ``c2``'s Toffsets are
+    ``{i: [1, 1], j: [0, 1]}``. Then, ``c2`` is translated w.r.t. ``c1`` by
+    ``(1, 0)``, and True is returned.
     """
-    assert len(ofs1) == len(ofs2)
-    return len({i - j for i, j in zip(ofs1, ofs2)}) == 1
+    assert len(c1.offsets) == len(c2.offsets)
+
+    # Transpose `offsets` so that
+    # offsets = [{x: 2, y: 0}, {x: 1, y: 3}] => {x: [2, 1], y: [0, 3]}
+    Toffsets1 = LabeledVector.transpose(*c1.offsets)
+    Toffsets2 = LabeledVector.transpose(*c2.offsets)
```

Comment:
i think if there's not particular reason for capitalizing it could be `toffsets1` and `toffsets2` instead